core/chains/evm/client/errors.go

```diff
@@ -167,13 +167,18 @@ var klaytn = ClientErrors{
 // Nethermind
 // All errors: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.TxPool/AcceptTxResult.cs
 // All filters: https://github.com/NethermindEth/nethermind/tree/9b68ec048c65f4b44fb863164c0dec3f7780d820/src/Nethermind/Nethermind.TxPool/Filters
-var nethermindFatal = regexp.MustCompile(`(: |^)(SenderIsContract|Invalid|Int256Overflow|FailedToResolveSender|GasLimitExceeded)$`)
+var nethermindFatal = regexp.MustCompile("(: |^)(SenderIsContract|Invalid(, transaction Hash is null)?|Int256Overflow|FailedToResolveSender|" +
+	"GasLimitExceeded(, Gas limit: [0-9]+, gas limit of rejected tx: [0-9]+)?|" +
+	"NonceGap(, Future nonce. Expected nonce: [0-9]+)?)$")
```

Comment:
Is `[0-9]+` the same as `\d+`?

Reply:
I found this:
`\d matches a digit (equivalent to [0-9])
`
so it looks like it's the same.

Reply:
In that case let's pick one to align on, maybe `\d` since it is simpler?

Reply:
Agree, although I would have to escape the `\` since it's in quotes. Eventually, it will be this `\\d+`

Reply:
oh that's too bad - what about using the other kind of quotes, '`', like `NonceTooLow` on 175?

Reply:
https://go.dev/ref/spec#String_literals "Raw string" that is

Reply:
I was afraid that backquotes always respect new lines (if there is a multiline string), but it turns out I can concat them in multiple lines (tests are passing). I'll change quotes to backquotes and `[0-9]` to `\d` .